server/utils/diningTableAlgo.js

```diff
@@ -11,75 +20,202 @@ exports.makeDiningTables = function(guests, numPerTable) {
     }
   };
 
-  // define number of tables
-  var numTables = Math.ceil(guests.length / numPerTable);
+  var canAddFriend = function(diningTable, friend1, friend2) {
+    var result = true;
 
```

Comment:
Would be helpful to add some comments in this function to explain in plain english what it is used for